src/Kunstmaan/ApiBundle/DependencyInjection/Compiler/TransformerCompilerPass.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Kunstmaan\ApiBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class TransformerCompilerPass implements CompilerPassInterface
```

Comment:
Would you consider using [mlebkowski/registry-compiler](https://github.com/mlebkowski/registry-compiler) for this? After registering this compiler pass in any bundle (see the readme) you would just tag the `kunstmaan_api.service.data_transformer` service:

```yaml
services:
  kunstmaan_api.service.data_transformer:
      tags: 
        - name: nassau.registry            # enable collecting related services / transformers
          tag: kunstmaan_api.transformer   # find them using this tag
          method: 'addTransformer'         # register them using this method
          order: priority                  # order them based on the "priority" attribute
          class: 'Kunstmaan\ApiBundle\Service\Transformers\TransformerInterface'
                                           # ensure they implement the interface at compile-time
```

And change the `addTransformer` method to accept `$index` as the first parameter, and `TransformerInterface $transformer` as the second.